src/benchmarks/real-world/Microsoft.ML.Benchmarks/Microsoft.ML.Benchmarks/HashBench.cs

```diff
@@ -0,0 +1,216 @@
+
+            public override bool IsColumnActive(int col)
+            {
+                if (col != 0)
+                    throw new Exception();
+                return true;
+            }
+
+            public override ValueGetter<TValue> GetGetter<TValue>(int col)
+            {
+                if (col != 0)
+                    throw new Exception();
+                if (_getter is ValueGetter<TValue> typedGetter)
+                    return typedGetter;
+                throw new Exception();
+            }
+
+            public static RowImpl Create<T>(ColumnType type, ValueGetter<T> getter)
+            {
+                if (type.RawType != typeof(T))
+                    throw new Exception();
+                return new RowImpl(type, getter);
+            }
+
+            private RowImpl(ColumnType type, Delegate getter)
+            {
+                var builder = new SchemaBuilder();
+                builder.AddColumn("Foo", type, null);
+                Schema = builder.GetSchema();
+                _getter = getter;
+            }
+        }
+
+        private const int Count = 100_000;
+
+        private readonly IHostEnvironment _env = new MLContext();
+
+        private RowImpl _inRow;
+        private ValueGetter<uint> _getter;
+        private ValueGetter<VBuffer<uint>> _vecGetter;
+
+        private void InitMap<T>(T val, ColumnType type, int hashBits = 20, ValueGetter<T> getter = null)
+        {
+            if (getter == null)
+                getter = (ref T dst) => dst = val;
+            _inRow = RowImpl.Create(type, getter);
+            // One million features is a nice, typical number.
+            var info = new HashingTransformer.ColumnInfo("Bar", "Foo", hashBits: hashBits);
+            var xf = new HashingTransformer(_env, new[] { info });
+            var mapper = xf.GetRowToRowMapper(_inRow.Schema);
+            var column = mapper.OutputSchema["Bar"];
+            var outRow = mapper.GetRow(_inRow, c => c == column.Index);
+            if (type is VectorType)
+                _vecGetter = outRow.GetGetter<VBuffer<uint>>(column.Index);
+            else
+                _getter = outRow.GetGetter<uint>(column.Index);
+        }
+
+        /// <summary>
+        /// All the scalar mappers have the same output type.
+        /// </summary>
+        private void RunScalar()
+        {
+            uint val = default;
+            for (int i = 0; i < Count; ++i)
+            {
+                _getter(ref val);
+                ++_inRow.PositionValue;
+            }
+        }
+
+        private void InitDenseVecMap<T>(T[] vals, PrimitiveType itemType, int hashBits = 20)
+        {
+            var vbuf = new VBuffer<T>(vals.Length, vals);
+            InitMap(vbuf, new VectorType(itemType, vals.Length), hashBits, vbuf.CopyTo);
+        }
+
+        /// <summary>
+        /// All the vector mappers have the same output type.
+        /// </summary>
+        private void RunVector()
```

Comment:
@najeeb-kazmi @TomFinley could you please explain what are we benchmarking here?

I just want to have a good understanding of why we want to add it to the performance repo and if we ever regress it, what does it mean?

@najeeb-kazmi most probably it would be good to add an xml summary doc comment to every ML benchmark class so perf folks like me can have better understanding of what we are benchmarking here

Reply:
@najeeb-kazmi yes, please. Could you add xml summary doc to the ML benchmarks?